lib/syntax-extension.rkt

```diff
@@ -0,0 +1,15 @@
+#lang racket/base
```

Comment:
I don't like the idea of having a file like this (defining new useful tools) in this package (which, before this PR, just collects existing useful tools).

Can you submit `define-simple-id-macro` and `define-pattern-expander` to the `syntax/parse/define` library?

Reply:
The weird thing about it is the phase. For it to be useful it needs to be required at phase 1, where normally you only have to require `syntax/parse/define` at phase 0.

Reply:
That's a good reason for agile to provide it.

But I still don't think agile should define it, so I don't want to merge this pull request as-is. (If agile defines this, then someone might want to `(require agile)` and agile will need more documentation to describe the useful things it defines.)

Reply:
Okay, that makes sense.